File: backup/Invest-Shop/Invest-Shop-Repository/src/main/java/com/illunex/invest/license/entity/PluginRole.java
package com.illunex.invest.license.entity;

import com.illunex.invest.license.enumable.RoleType;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class PluginRole {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long pluginRoleIdx;
    RoleType role;
}

File: microservices/shop/src/main/kotlin/com/illunex/invest/shop/persistence/entity/PluginRole.kt
package com.illunex.invest.shop.persistence.entity

import javax.persistence.*

@Entity
@Table(name = "plugin_role")
class PluginRole {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var roleTitle: String = ""
    var menuTitle: String = ""

    constructor()

    constructor(roleTitle: String) {
        this.roleTitle = roleTitle
    }
}
